### DISCUSSION TITLE
How to get spreadsheet cell values as xsd:string (like displayed, ignoring the cell formatting)?

### DISCUSSION THREAD

**Discussion by sgoetz-brox** [SPARQL-Anything/sparql.anything#481] [Q&A]
When entering these values in an XLSX file (without manually changing the cell formatting; in a locale that uses `,` as decimal separator):

```
foo
23
23.0
23.42
23,42
2024-05-29
29.05.24
```

I’m getting this output with SPARQL Anything:

```
"foo"
"23.0"^^xsd:double
"23.0"
"23.42"
"23.42"^^xsd:double
"45441.0"^^xsd:double
"45441.0"^^xsd:double
```

Instead, I would like to get everything as string (`xsd:string`), just like it’s _displayed_ in Excel/LibreCalc, i.e.:

```
"foo"
"23"
"23.0"
"23.42"
"23,42"
"2024-05-29"
"29.05.24"
```

When changing the formatting of a column to "Text" and entering the values again, I’m getting the desired output. But when converting an existing column, it still seems to convert the two dates into (I suppose) Excel’s own number representation. Anyway, I would like to avoid changing the input file, if possible.

**Comment by luigi-asprino** [comment]
Hi @sgoetz-brox,

At the moment the engine automatically maps primitive Java types onto xsd datatypes. 
An option `interpret-values-as-literals` can be introduced to accommodate your request (I suppose).
Maybe rdfs:Literal is more appropriate than xsd:string.
This could be a general option (i.e. not specific for spreadsheet data sources) enabling users to manipulate raw values freely.

**Reply by sgoetz-brox** [reply]
Hi @luigi-asprino, thanks for the quick reply :)

> An option `interpret-values-as-literals` can be introduced to accommodate your request
>
> […]
>
> This could be a general option (i.e. not specific for spreadsheet data sources) enabling users to manipulate raw values freely.

I think that would be great!

Would this option also allow getting the date (as displayed: `2024-05-29` / `29.05.24`) instead of Excel’s serial number representation of the date (`45441`)?  Or would this option merely prevent that SPARQL Anything makes this serial number an `xsd:double`, so the output would still be the serial number (but now as string)?

**Reply by luigi-asprino** [reply]
With #482 dates should be interpreted correctly.
